frontend/src/libs/packages/store/middlewares/middlewares.ts

```diff
@@ -0,0 +1,27 @@
+import { type AnyAction, type Middleware } from '@reduxjs/toolkit';
+import { isRejected } from '@reduxjs/toolkit';
+
+import { NotificationType } from '#libs/packages/notification/libs/enums/enums.js';
+import { type AppDispatch } from '#libs/packages/store/libs/types/types.js';
+import { appActions } from '#slices/app/app-notification.js';
+
+const handleError: Middleware<unknown, unknown, AppDispatch> = ({
+  dispatch,
+}) => {
+  return (next) => {
+    return (action: AnyAction) => {
+      if (isRejected(action)) {
```

Comment:
```suggestion
      if (isRejected(action) && action.error) {
```

Reply:
I previously had it set up in the same manner, but without the conditional part (**?? 'Something went wrong'**). The issue is that the ESLint rules don't permit both configurations:

At line 14, column 22, there's an error flagged: "Unnecessary conditional," with the message advising the left-hand side of the ?? operator to potentially be null or undefined, as per the @typescript-eslint/no-unnecessary-condition rule.
Because we introduced the conditional check '**&& action.error**,' we will never utilize the ?? 'Something went wrong' part. TypeScript interprets it as an unnecessary condition.

Could you please let me know which solution you would prefer me to implement?

Reply:
let's use it without `?? 'Something went wrong'` then

Reply:
I previously had it set up in the same manner, but without the conditional part (?? 'Something went wrong'). The issue is that the ESLint rules don't permit both configurations:

  13:33  error  Unnecessary conditional, value is always truthy  @typescript-eslint/no-unnecessary-condition
Because we introduced the conditional check '&& action.error,' we will never utilize the ?? 'Something went wrong' part. TypeScript interprets it as an unnecessary condition.

Could you please let me know which solution you would prefer me to implement?

Reply:
i will update this, i saw your comment after committing.

Reply:
Hi, I'm back again! After all the tweaks we've made, if I take out "action.error," I get this error message:

Type 'string | undefined' is not assignable to type 'string'.
 Type 'undefined' is not assignable to type 'string'.ts(2322)
notification-payload.type.ts(7, 3): The expected type comes from property 'message' which is declared here on type 'NotificationPayload'

We're expecting a string, but the message might sometimes be undefined. To fix this, one way is to change "&& action.error" to "&& action.error.message." or do a triple-check. If the message is there, it'll pop up. However, if there's a glitch without a message, we won't catch it.

Now, about "&& action.error": if we keep it and the error message doesn't exist, we'll end up with an empty space. In this case, I'm thinking it might be better to stick with the "?? 'Something went wrong'" condition and drop the "&& action.error." If something goes wrong and the action gets rejected without triggering an "action.error," at least we'll display "Something went wrong" so users know there's an issue. Plus, if the error does get added to the rejection later on, we'll be all set to show the error message.

Of course, that's just my opinion. I'm sure there are other ways to go about it, but we're kinda of limited by the linter and type restrictions. For now, I'll go ahead and push the code like this. Take your time to read through this and let me know which solution you're leaning toward, or if you have a better solution let me know and I will implement it as soon as possible.